rdkit/Chem/fmcs/fmcs.py

```diff
@@ -674,24 +673,23 @@ def _check_atom_classes(molno, num_atoms, atom_classes):
 # prune. But so far I don't have a test set which drives the need for
 # that.
 
+# UPDATED by Ichiru Take: Use collections.Counter to count the bond types, which is much faster.
+# See here: https://stackoverflow.com/questions/44012479/intersection-of-two-counters
 
 # Return a dictionary mapping iterator item to occurrence count
 def get_counts(it):
-  d = defaultdict(int)
-  for item in it:
-    d[item] += 1
-  return dict(d)
-
+  return dict(Counter(it))
 
 # Merge two count dictionaries, returning the smallest count for any
 # entry which is in both.
 def intersect_counts(counts1, counts2):
-  d = {}
-  for k, v1 in counts1.iteritems():
-    if k in counts2:
-      v = min(v1, counts2[k])
-      d[k] = v
-  return d
+  if not isinstance(counts1, Counter) and not isinstance(counts2, Counter):
+    return dict(Counter(counts1) & Counter(counts2))
+  if isinstance(counts1, Counter) and not isinstance(counts2, Counter):
+    return dict(counts1 & Counter(counts2))
+  if not isinstance(counts1, Counter) and isinstance(counts2, Counter):
+    return dict(Counter(counts1) & counts2)
+  return dict(counts1 & counts2)
```

Comment:
With all the if statements this is quite hard to read. This should be equivalent.
```
If not isinstance(Counter, counts1):
  Counts1 = Counter(counts1)
If not isinstance(Counter, counts2):
  Counts2 = Counter(counts2)
Return dict(counts1 & counts2)
```
The question is however, is it any faster if we have to add two conversions from dict to counter and one for counter to dict?

I looked at the implementation of the & operation and it is done in Python. I doubt that it is any faster than the implementation we currently have. I expect it to be slower

Reply:
I really think this needs to be reversed to the original code. It will be slower.

Reply:
OK, I got your idea! I would fix it in the next patch. Thank you for your review.